### ISSUE TITLE
Average Energy Consumption - choice between Wh/km and km/kWh

### ISSUE THREAD

**Issue by Symbiot78** [Hyundai-Kia-Connect/kia_uvo#368]
**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
The choice between Wh/km & km/kWh

**Describe alternatives you've considered**
som sort of template string conversion in a lovelace card.

I've never heard anyone use wH/km when talking about EV consumption..

**Comment by adam-cobb** [comment]
Yes please. When I view the sensor it says it's unit_of_measurement: Wh/mi but it turns out it is actually Wh/km by the looks of it

**Comment by adam-cobb** [comment]
> Yes please. When I view the sensor it says it's unit_of_measurement: Wh/mi but it turns out it is actually Wh/km by the looks of it

I have actually created a template sensor that does 1000 / 1.609344 / <average energy consumption sensor value> to convert it to miles/kWh (which is my preference), so not a big issue

**Comment by Symbiot78** [comment]
@adam-cobb  - share or it didn’t happen 😅😅

**Comment by rsegers** [comment]
The consumption is calculated using a value provided by the API named `calculativeOdo` . I would expect that this value follows the user's preference for either miles or kilometers. Apparently, it does not? If that's the case, calculation should convert it to the preference.

Regarding the discussion Wh/km or km/kWh: the second makes more sense to me. But for EV's the first (Wh/km) is more common.
